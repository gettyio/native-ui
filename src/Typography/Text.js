import styled from 'styled-components/native'

const Text = styled.Text`
  font-family: ${props =>
    props.font || props.theme.typography[props.type].font};
  font-size: ${props =>
    props.size || props.theme.typography[props.type].size}px;
  font-weight: ${props =>
    props.weight || props.theme.typography[props.type].weight};
  line-height: ${props =>
    props.leading || props.theme.typography[props.type].leading}px;
  letter-spacing: ${props =>
    props.tracking || props.theme.typography[props.type].tracking}px;
  color: ${props => props.color || props.theme.typography[props.type].color};
`

Text.defaultProps = {
  type: 'body',
  theme: {
    typography: {
      body: {
        font: 'sans-serif',
        weight: 500,
        size: 22,
        leading: 28,
        tracking: -0.384,
        color: 'black'
      }
    }
  }
}

export default Text
