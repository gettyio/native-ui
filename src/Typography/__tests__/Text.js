import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import 'jest-styled-components'

import Text from '../Text'

describe('Text component', () => {
  it('should render without issue', () => {
    const component = shallow(<Text />)

    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })
})
