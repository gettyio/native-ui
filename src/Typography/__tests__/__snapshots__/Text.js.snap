// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should render without issue 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "black",
        "fontFamily": "sans-serif",
        "fontSize": 22,
        "fontWeight": "500",
        "letterSpacing": -0.384,
        "lineHeight": 28,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "typography": Object {
        "body": Object {
          "color": "black",
          "font": "sans-serif",
          "leading": 28,
          "size": 22,
          "tracking": -0.384,
          "weight": 500,
        },
      },
    }
  }
  type="body"
/>
`;
